dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -483,6 +486,11 @@ private Object parseFlatField(FieldDef fd, boolean isLittleEndian) throws Unpars
                     codePoints = ((int) getInt(4, isLittleEndian));
                 }
 
+                if (codePoints > MAX_STRING_LENGTH)
+                {
+                    throw new UnparsableContentException("String length exceeds maximum supported length: " + MAX_STRING_LENGTH);
```

Comment:
if we have the field name or position, and the reported string len, let's report them. Also, as written, it's not clear who is imposing the length restriction (JAVA?).